component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/extension/AfterVariableMetadataEnricher.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (C) 2006-2021 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.runtime.manager.extension;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.talend.sdk.component.api.component.AfterVariables.AfterVariable;
+import org.talend.sdk.component.api.input.Emitter;
+import org.talend.sdk.component.api.input.PartitionMapper;
+import org.talend.sdk.component.api.processor.Output;
+import org.talend.sdk.component.spi.component.ComponentMetadataEnricher;
+
+/**
+ * Search annotation {@link AfterVariable} and add a new meta information about description of after variables.
+ */
+public class AfterVariableMetadataEnricher implements ComponentMetadataEnricher {
+
+    private static final Set<Class<? extends Annotation>> SUPPORTED_ANNOTATIONS = new HashSet<>();
+
+    private static final String META_KEY_AFTER_VARIABLE = "variables::after::value";
+
+    private static final String VALUE_DELIMITER = "\\:";
+
+    private static final String LINE_DELIMITER = "\\;";
+
+    static {
+        SUPPORTED_ANNOTATIONS.add(PartitionMapper.class);
+        SUPPORTED_ANNOTATIONS.add(Emitter.class);
+        SUPPORTED_ANNOTATIONS.add(Output.class);
+    }
```

Comment:
You can init the set using `new HashSet<>(Arrays.asList(PartitionMapper.class, Emitter.class, ...) )` instead of static block.

Reply:
Nice idea, thank you!